lib/views/course_detail_view/detail_course_List.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:gocast_mobile/views/components/base_view.dart';
+import 'package:gocast_mobile/views/course_detail_view/stream_card.dart';
+
+class DetailCourseList extends ConsumerWidget {
+  final String title;
+  final List<StreamCard> streamCard;
+  final Future<void> Function()? onRefresh;
+
+  const DetailCourseList({
+    super.key,
+    required this.title,
+    required this.streamCard,
+    this.onRefresh,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return BaseView(
+      title: title,
+      actions: [
+        IconButton(
+          icon: const Icon(Icons.search),
```

Comment:
I think it looks pretty good, I haven't come across any functional flaws or bugs. I just have a few suggestions:
- The navigation bar should look like in the Figma (search input is not hidden, back arrow icon etc.), for example, it should look like this:
<img width="230" alt="Screenshot 2023-12-27 at 21 13 10" src="https://github.com/TUM-Dev/gocast_mobile/assets/43902289/85b2e1e1-d6d8-4e3e-899d-e3f179ca90a1">